python/console/console_settings.ui

```diff
@@ -433,13 +367,210 @@
+               <string>From doc and APIs</string>
+              </property>
+             </widget>
+            </item>
+            <item row="0" column="1">
+             <widget class="QRadioButton" name="autoCompFromAPI">
+              <property name="toolTip">
+               <string>Get autocompletion from installed APIs</string>
+              </property>
+              <property name="text">
+               <string>From API files</string>
+              </property>
+              <property name="checked">
+               <bool>true</bool>
+              </property>
+             </widget>
+            </item>
+            <item row="0" column="0">
+             <widget class="QRadioButton" name="autoCompFromDoc">
+              <property name="toolTip">
+               <string>Get autocompletion from current document</string>
+              </property>
+              <property name="text">
+               <string>From document</string>
+              </property>
+             </widget>
+            </item>
+           </layout>
+          </item>
+         </layout>
+        </widget>
+       </item>
+       <item row="2" column="0">
+        <widget class="QgsCollapsibleGroupBox" name="groupBoxFromatting">
+         <property name="title">
+          <string>Formatting</string>
+         </property>
+         <property name="collapsed" stdset="0">
+          <bool>false</bool>
+         </property>
+         <property name="saveCollapsedState" stdset="0">
+          <bool>true</bool>
+         </property>
+         <layout class="QGridLayout" name="gridLayout_4">
+          <property name="verticalSpacing">
+           <number>2</number>
+          </property>
+          <item row="2" column="1">
+           <widget class="QSpinBox" name="maxLineLength">
+            <property name="minimum">
+             <number>60</number>
+            </property>
+            <property name="maximum">
+             <number>200</number>
+            </property>
+            <property name="value">
+             <number>80</number>
+            </property>
+           </widget>
+          </item>
+          <item row="0" column="0" colspan="2">
+           <widget class="QCheckBox" name="formatOnSave">
+            <property name="text">
+             <string>Reformat on save</string>
+            </property>
+           </widget>
+          </item>
+          <item row="5" column="0" colspan="2">
+           <widget class="QCheckBox" name="blackNormalizeQuotes">
+            <property name="text">
+             <string>Normalize quotes</string>
+            </property>
+           </widget>
+          </item>
+          <item row="3" column="1">
+           <widget class="QComboBox" name="formatter">
+            <property name="minimumSize">
+             <size>
+              <width>100</width>
+              <height>0</height>
+             </size>
+            </property>
+            <item>
+             <property name="text">
```

Comment:
I have a slight preference to populating combo boxes like this in code, rather than ui files -- (it makes it easier to find references in IDEs)